From b22250cf496a0e8e44e6d442417818dff1386522 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 5 Jan 2021 14:19:03 +0200
Subject: [PATCH 1055/1921] net: mvpp2: fix driver deffer with legacy device
 tree

Change-Id: I67fa84299d5d24686a23961829a4d848183d26b0
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/43034
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 3bc39c0f2f54..4f66dcfb0eaf 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -7636,7 +7636,7 @@ static int mvpp2_probe(struct platform_device *pdev)
 err_pp_clk:
 	clk_disable_unprepare(priv->pp_clk);
 err_cm3:
-	if (!has_acpi_companion(&pdev->dev))
+	if (!has_acpi_companion(&pdev->dev) && priv->cm3_base)
 		gen_pool_free(priv->sram_pool, (unsigned long)priv->cm3_base,
 			      MSS_SRAM_SIZE);
 
-- 
2.31.1

